src/components/package-port/CodeAndPreview.tsx

```diff
@@ -346,7 +308,16 @@ export function CodeAndPreview({ pkg }: Props) {
     })
   }
 
-  const { handleCreateFile } = useFileManagement(state, setState)
+  const { handleCreateFile, handleDeleteFile, handleSave, fsMap } =
+    useFileManagement({
+      setCodeAndPreviewState: setState,
+      pkg,
+      updatePackageFilesMutation,
```

Comment:
bad API name (always start with `on*` for event handlers). Don't pass mutations (always a leaky abstraction), the handling must be where the mutation is created.

```suggestion
      onPackageFileCreated,
```